### PR TITLE
Fix max measurements gauge test

### PR DESCRIPTION
**Pull request by ccapurso**
We have a theory that the amount of metrics emitted in `TestGauge_MaximumMeasurements` is causing the test to timeout in CircleCI. It seems like we can exercise the same functionality with a fraction of the original metric count. A constant for the batch size has also been added for clarification when debugging the test and associated underlying  `streamGaugesToSink` function.

### CODE REVIEW

File: helper/metricsutil/gauge_process.go [hunk 1]
```diff
@@ -197,16 +197,18 @@ func (p *GaugeCollectionProcess) collectAndFilterGauges() {
 	p.streamGaugesToSink(values)
 }
 
+// batchSize is the number of metrics to be sent per tick duration.
+const batchSize = 25
```
Comment (swayne275) [inline helper/metricsutil/gauge_process.go:201]: lovely

File: helper/metricsutil/gauge_process.go [hunk 2]
```diff
@@ -197,16 +197,18 @@ func (p *GaugeCollectionProcess) collectAndFilterGauges() {
...
+
 func (p *GaugeCollectionProcess) streamGaugesToSink(values []GaugeLabelValues) {
 	// Dumping 500 metrics in one big chunk is somewhat unfriendly to UDP-based
```
Comment (swayne275) [inline helper/metricsutil/gauge_process.go:204]: i know this was already here, but where is the 500 coming from?

**Review by swayne275 — APPROVED** [review]
nice cleanup! big question about a comment that was already there though (and seems to be the basis of the 25 items per tick

File: helper/metricsutil/gauge_process.go (hunk 2 above)
Comment (hghaf099) [reply helper/metricsutil/gauge_process.go:204]: Yeah, I have the same question, cause that essentially affects the batchsize

Comment (hghaf099) [reply helper/metricsutil/gauge_process.go:204]: seems to be from here:
https://github.com/hashicorp/vault/blob/5f590d4d505dcb7f0021893c23ce2ab49f204d11/internalshared/configutil/telemetry.go#L29

File: helper/metricsutil/gauge_process.go (hunk 1 above)
Comment (hghaf099) [reply helper/metricsutil/gauge_process.go:201]: I think this should depend on the default value here:
https://github.com/hashicorp/vault/blob/5f590d4d505dcb7f0021893c23ce2ab49f204d11/internalshared/configutil/telemetry.go#L29

Comment (ccapurso) [reply helper/metricsutil/gauge_process.go:201]: I just lifted the local var (`batchSize := 25`) from `streamGuagesToSink` and made it a package-level const.

File: helper/metricsutil/gauge_process.go (hunk 2 above)
Comment (ccapurso) [reply helper/metricsutil/gauge_process.go:204]: 500 was the original `MaxGaugeCardinality` for the sink in the test. We reduced that to 100.

See:
https://github.com/hashicorp/vault/blob/7fd37fe778eb9ff187beb9af023d5fa1e5f50360/helper/metricsutil/gauge_process_test.go#L436

Comment (hghaf099) [reply helper/metricsutil/gauge_process.go:204]: I think we may need to think what would happen if `MaximumGaugeCardinalityDefault` configured differently that the default value.

Comment (ccapurso) [reply helper/metricsutil/gauge_process.go:204]: I believe that `MaximumGaugeCardinalityDefault` comes into play if the telemetry configuration doesn't specify `MaximumGaugeCardinality`. It isn't configurable.

https://github.com/hashicorp/vault/blob/5f590d4d505dcb7f0021893c23ce2ab49f204d11/internalshared/configutil/telemetry.go#L212-L214

Comment (hghaf099) [reply helper/metricsutil/gauge_process.go:204]: Yes, sorry I am not talking about the test though. The test is perfectly fine. 
I am saying it seems the calculations in this function seem to be all based on the assumption that `MaximumGaugeCardinality` is always 500 which is the default value. So, if that is configured differently, do we want to keep the constant values in this function as is? `MaximumGaugeCardinality` could be configured to be 1000 for example.

Comment (ccapurso) [reply helper/metricsutil/gauge_process.go:204]: Oh that's an interesting point! Attempting to fix this test together has brought up some interesting discussion. It looks like `MaximumGaugeCardinality ` is used in the calling function `collectAndFilterGauges`. Having a constant batch size that is not calculated based on the `MaxGaugeCardinality` means that transmission rate does not scale as the max increases. There obviously needs to be a max batch size to address the concerns discussed in the comment but your point is starting to make me wonder if this hard-coded batch size is warranted or if any flexibility can be introduced. I'm going to dig into the logic a bit more closely now that you mentioned this.

Comment (ccapurso) [reply helper/metricsutil/gauge_process.go:204]: Yup, I agree. I think we should do that as a separate PR though. I'd rather allow the necessary time to investigate it more thoroughly and provide appropriate test coverage. I can make a ticket with context from our discussion here.

**Pull request merged**